### PR TITLE
docs: warn about missing docs for `jj-lib` crate

### PR DESCRIPTION
**Pull request by arxanas**
Should we add anything else to the docstring for the `jj-lib` crate?

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Review by martinvonz — APPROVED** [review]
I think this seems like a good idea (as I think I said on the fsmonitor PR). I'm not used to adding documentation to everything, so I expect that this will be a little annoying to get used to, but it's probably a good habit to form. I'm curious if others have experience with enabling this and how well it works in practice.

File: lib/src/lib.rs [hunk 1]
```diff
@@ -12,9 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#![deny(unused_must_use)]
 //! Jujutsu version control system.
 
+#![deny(missing_docs, unused_must_use)]
```
Comment (yuja) [inline lib/src/lib.rs:17]: Maybe `warn(missing_docs)` is better for local development? I tend to leave draft patches without docs. I think CI should deny it, though.

Comment (arxanas) [reply lib/src/lib.rs:17]: @yuja do you think we should also adjust `unused_must_use` to `warn`?

**Comment by arxanas** [comment]
> I'm curious if others have experience with enabling this and how well it works in practice.

@martinvonz it's enabled for git-branchless and it works for me, although I don't work with others too much on it 🙂. The documentation is only required for `pub` items, so I think it ends up being required mostly in places where it makes sense, rather than mostly implementation details.

**Pull request merged**